pkg/api/addtask.go

```diff
@@ -0,0 +1,119 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go1f/pkg/db"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// addTaskHandler обрабатывает POST-запросы для создания задач
+// Проверяет входные данные и сохраняет задачу в БД
+func addTaskHandler(w http.ResponseWriter, r *http.Request) {
+	// Декодирование JSON тела запроса
+	var task db.Task
+	if err := json.NewDecoder(r.Body).Decode(&task); err != nil {
+		writeJson(w, map[string]string{"error": "ошибка декодирования JSON"})
+		return
+	}
+	// Валидация обязательных полей
+	if task.Title == "" {
+		writeJson(w, map[string]string{"error": "не указан заголовок задачи"})
+		return
+	}
+	// Проверка и корректировка даты
+	now := time.Now()
+	if err := checkDate(&task, now); err != nil {
+		writeJson(w, map[string]string{"error": err.Error()})
+		return
+	}
+	// Сохранение задачи в БД
+	id, err := db.AddTask(&task)
+	if err != nil {
+		writeJson(w, map[string]string{"error": "ошибка сохранения задачи"})
+		return
+	}
+	// Отправка успешного ответа
+	writeJson(w, map[string]string{"id": strconv.FormatInt(id, 10)})
+}
+
+// checkDate корректирует дату задачи
+func checkDate(task *db.Task, now time.Time) error {
+	// Валидация правила повторения
+	if err := validateRepeatRule(task.Repeat); err != nil {
+		return err
+	}
+	// Обработка специальных значений даты
+	if task.Date == "today" {
+		task.Date = now.Format("20060102")
+	}
+
+	if task.Date == "" {
+		task.Date = now.Format("20060102")
+	}
+	// Парсинг даты выполнения
+	t, err := time.Parse("20060102", task.Date)
+	if err != nil {
+		return errors.New("неверный формат даты")
+	}
+	// Корректировка прошедших дат
+	if t.Before(now) {
+		if task.Repeat == "" {
+			// Для одноразовых задач - установка текущей даты
+			task.Date = now.Format("20060102")
+		} else {
+			// Для повторяющихся задач - расчет следующей даты
+			if t.Truncate(24 * time.Hour).Equal(now.Truncate(24 * time.Hour)) {
+				task.Date = t.Format("20060102")
+			} else {
+				next, err := NextDate(now, task.Date, task.Repeat)
+				if err != nil {
+					return fmt.Errorf("неверное правило повторения: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
					return fmt.Errorf("неверное правило повторения: %w", err)
```